src/MatrixFreeLowMachEquationSystem.C

```diff
@@ -121,17 +121,15 @@ void
 MatrixFreeLowMachEquationSystem::register_copy_state_algorithm(
   std::string name, int length, stk::mesh::Part& part)
 {
-  auto* field =
-    meta_.get_field<VectorFieldType>(stk::topology::NODE_RANK, name);
+  auto* field = meta_.get_field(stk::topology::NODE_RANK, name);
   auto copy = new CopyFieldAlgorithm(
-    realm_, &part, &field->field_of_state(stk::mesh::StateNP1),
-    &field->field_of_state(stk::mesh::StateN), 0, length,
-    stk::topology::NODE_RANK);
+    realm_, &part, field->field_state(stk::mesh::StateNP1),
+    field->field_state(stk::mesh::StateN), 0, length, stk::topology::NODE_RANK);
```

Comment:
So here you are switching from the Field<T> derived type to the FieldBase base class? Is this to fix the std::bad_cast error? I'm puzzled by that.

Reply:
Yeah, just using the base class here (`CopyFieldAlgorithm` just takes the base class anyway).  I was passing both fields of different derived types (`VectorFieldType` and `ScalarFieldType`) to this algorithm but casting to `VectorFieldType`.

Reply:
Interesting. So in this case clang was actually catching a bad cast that other compilers were ignoring?

Reply:
I think it's probably the address sanitizer being active that allowed it to be caught (I build with clang, though not clang-7).  Not sure why it wasn't a sanitizer error directly, though.